### PR TITLE
feat: add `starts_with` for dask

### PR DESCRIPTION
**Pull request by aidoskanapyanov**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #637

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: narwhals/_dask/namespace.py [hunk 1]
```diff
@@ -28,6 +34,24 @@ class DaskNamespace:
     def __init__(self, *, backend_version: tuple[int, ...]) -> None:
         self._backend_version = backend_version
 
+    def _create_expr_from_callable(
```
Comment (MarcoGorelli) [inline narwhals/_dask/namespace.py:37]: where is this used?

File: narwhals/_dask/expr.py [hunk 2]
```diff
@@ -34,6 +35,11 @@ def __init__(
         self._output_names = output_names
         self._backend_version = backend_version
 
+    def __narwhals_namespace__(self) -> DaskNamespace:
+        from narwhals._dask.namespace import DaskNamespace
+
+        return DaskNamespace(backend_version=self._backend_version)
```
Comment (MarcoGorelli) [inline narwhals/_dask/expr.py:41]: where is this used?

**Review by MarcoGorelli — COMMENTED** [review]
thanks @aidoskanapyanov !

implementation looks fine, CI is just failing because of a couple of methods you've added which aren't used yet

could you remove them for now please?

**Comment by aidoskanapyanov** [comment]
Yep, I forgot to remove these, I sort of using `PandasLikeExpr` and `PandasLikeNamespace` as a reference.

**Review by MarcoGorelli — APPROVED** [review]
nice one @aidoskanapyanov feel free to merge!

**Pull request merged**